easybuild/easyblocks/o/openblas.py

```diff
@@ -94,3 +168,132 @@ def sanity_check_step(self):
             'dirs': [],
         }
         super(EB_OpenBLAS, self).sanity_check_step(custom_paths=custom_paths)
+
+    def _optarch_for_compiler(self, optarch):
+        """
+        Extracts the appropriate optarch for the compiler currently being used.
+        If it is not compiler-specific it is returned as-is.
+        If no optarch is found, False is returned.
+        :param optarch: A complete optarch statement.
+        https://easybuild.readthedocs.io/en/latest/Controlling_compiler_optimization_flags.html
+        """
+        if optarch is False:
+            return False
+
+        compiler = self.toolchain.comp_family()
+        compiler_specific_optarch_string = ''
+
+        if type(optarch) == str:
+            compiler_specific_optarch_string = optarch
+        elif type(optarch) == dict:
+            if compiler in optarch:
+                compiler_specific_optarch_string = optarch[compiler]
+        else:
+            raise EasyBuildError("optarch in an unexpected format: '%s'" % type(optarch)).__class__.__name__
+
+        return compiler_specific_optarch_string
+
+    def _parse_optarch(self, compiler_optarch):
+        """
+        Pick the march out of a given optarch.
+        Note that the result is lowercased.
+        :param compiler_optarch: An optarch for a given compiler.
+        https://easybuild.readthedocs.io/en/latest/Controlling_compiler_optimization_flags.html
+        """
+
+        target_arch = ''
+        pieces = compiler_optarch.split()
+
+        for piece in pieces:
+            spec = piece.split('=')
+            if spec[0] == 'march' or spec[0] == '-march':
+                target_arch = spec[1]
+
+        return target_arch.lower()
+
+    def _get_openblas_targets(self, targetfile):
```

Comment:
For all those functions: Could you make them free functions? In PyTorch/TensorFlow I found it a good idea so you can just import this from regular python and run/test it

Reply:
@lexming (I think) suggested generalising a number of these functions and shipping them off to Framework. The ones that are OpenBLAS specific should probably suffer a similar fate. This will require another rewrite though, not sure about the ETA on that.